debezium-core/src/main/java/io/debezium/relational/TableEditorImpl.java

```diff
@@ -92,47 +96,30 @@ public TableEditor setColumns(Column... columns) {
     public TableEditor setColumns(Iterable<Column> columns) {
         sortedColumns.clear();
         addColumns(columns);
-        updatePrimaryKeys();
         assert positionsAreValid();
         return this;
     }
 
     protected void updatePrimaryKeys() {
-        // table does not have any primary key, no need to update
-        if (uniqueValues) {
-            return;
-        }
-        Iterator<String> nameIter = this.pkColumnNames.iterator();
-        while (nameIter.hasNext()) {
-            String pkColumnName = nameIter.next();
-            if (!hasColumnWithName(pkColumnName)) {
-                nameIter.remove();
-            }
+        if (!uniqueValues) {
+            // table does have any primary key --> we need to remove it
+            this.pkColumnNames.removeIf(pkColumnName -> {
+                boolean pkColumnDoesNotExists = !hasColumnWithName(pkColumnName);
+                if (pkColumnDoesNotExists) {
+                    LOGGER.warn("The column \"" + pkColumnName + "\" is referenced as PRIMARY KEY, but a matching column is not defined in table \"" + tableId() + "\"!");
```

Comment:
I decided to just log an error here. But maybe we can also throw the previous IllegalArgumentException here. This should not happen from a proper captured DB schema, the reason can either be a bug in callers or a bug in the DDL parser logic. WDYT?

Reply:
@rk3rn3r Could we reuse `inconsistent.schema.handling.mode` for this?

Reply:
@jpechane I have some updated pov about it today. I'm still not sure if there's any case this can happen with a valid, captured db schema on any of our connectors (I can't imagine any weird case that's causing it, but this might be an edge case)? So imo this can only happen when the parsing logic is broken. The only question I have is, that even if parts of the logic is broken, can we successfully continue? My today's conclusion is that it's not safe to leave the connector running and an exception is the best way to handle it now. It will make DDL parsing errors more visible.
If you don't have any objections, I will switch to the exception.

Reply:
I agree, I beleive we are done, the PR can be merged when CI is finished.